library/ceph_volume_zap.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/python
+import datetime
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.0',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: ceph_volume_zap
+
+short_description: Zap devices used by ceph OSDs with ceph-volume
+
+description:
+    - Using the ceph-volume utility available in Ceph this module
+      can be used to zap devices used by ceph OSDs.
+    - These devices, partitions or logicial volumes can then be reused.
+    - All partitions and logical volumes will be kept intact.
+
+options:
+    device:
+        description:
+            - The logical volume name, device or partition to zap.
+        required: true
+    device_vg:
+        description:
+            - If device is a lv, this must be the name of the volume group it belongs to.
+        required: false
+    destroy:
```

Comment:
Perhaps we could call this `destroy_lv` or `destroy_data`? The distinction between zap and destroy is not clear so using a zap module with a destroy option is not a bit weird. So we need to clarify that.

Reply:
I understand this flag can be confusing, I've only named it ``destroy`` so it is the same as the ``--destroy`` flag of ``ceph-volume lvm zap``.  If ``lvm zap`` is run against an lv then ``--destroy`` is a noop. The ``-destroy`` flag is used when zapping a partition or raw device that ceph-volume has previously used and created lvs on. This was introduced because users were providing a raw device for ``--data`` and needed an easy way to remove the lvs that ceph-volume creates in that situation.

I'm not really against changing the name of the flag, but maybe we can clarify better with the documentation? It might be confusing if it's not named the same as the ``ceph-volume lvm zap`` flag too.